### PR TITLE
Hotfix: fix plugins option in config file

### PR DESCRIPTION
**Pull request by oxarbitrage**
This is a replacement for https://github.com/bitshares/bitshares-core/pull/1640 as we are going top create a hotfix for this issue to merge as soon as possible into `master`.

This is for issue https://github.com/bitshares/bitshares-core/issues/1637

### CODE REVIEW

**Comment by abitmore** [comment]
I'm fine with this. @pmconrad your opinion?

**Comment by pmconrad** [comment]
Hm, `witness_node --help` no longer reports the plugin options, right?

**Comment by oxarbitrage** [comment]
Nice catch @pmconrad . I fixed at https://github.com/bitshares/bitshares-core/pull/1647/commits/de22e990499e974bf6dab2df8aef1132428c137a

Help command need to be moved to after the plugins are enabled and also `app_options` populated again.

Thanks.

**Comment by abitmore** [comment]
My bad. I guess `bpo::parse_command_line` also need plugins?

**Comment by oxarbitrage** [comment]
You are right @abitmore , the plugin options were giving error when called from the command line.
The command line parsing should be done only once, but after the plugin options are in, so i moved the try/catch block with the parsing to the second place where we have all the options.

Let me know how do you see it, it is not very clean for sure but seems to be working.

File: programs/witness_node/main.cpp [hunk 1]
```diff
@@ -83,49 +103,26 @@ int main(int argc, char** argv) {
...
 
+      // add plugin options to config
       try
       {
          bpo::options_description cli, cfg;
          node->set_program_options(cli, cfg);
          app_options.add(cli);
          cfg_options.add(cfg);
-         bpo::store(bpo::parse_command_line(argc, argv, app_options), options);
+	 bpo::store(bpo::parse_command_line(argc, argv, app_options), options);
```
Comment (abitmore) [inline programs/witness_node/main.cpp]: Please revert this white-space change.

**Comment by abitmore** [comment]
Duplicate options in output of `--help` command:
```
$ ./witness_node --help 
Graphene Witness Node:
  -h [ --help ]                         Print this help message and exit.
  -d [ --data-dir ] arg (="witness_node_data_dir")
                                        Directory containing databases, 
                                        configuration file, etc.
  -v [ --version ]                      Display version information
  --plugins arg (=witness account_history market_history grouped_orders)
                                        Space-separated list of plugins to 
                                        activate

  --replay-blockchain                   Rebuild object graph by replaying all 
                                        blocks without validation
  --revalidate-blockchain               Rebuild object graph by replaying all 
                                        blocks with full validation
  --resync-blockchain                   Delete all blocks and re-sync with 
                                        network from scratch
  --force-validate                      Force validation of all transactions 
                                        during normal operation
  --genesis-timestamp arg               Replace timestamp from genesis.json 
                                        with current time plus this many 
                                        seconds (experts only!)

  --p2p-endpoint arg                    Endpoint for P2P node to listen on
  -s [ --seed-node ] arg                P2P nodes to connect to on startup (may
                                        specify multiple times)
  --seed-nodes arg                      JSON array of P2P nodes to connect to 
                                        on startup
  -c [ --checkpoint ] arg               Pairs of [BLOCK_NUM,BLOCK_ID] that 
                                        should be enforced as checkpoints.
  --rpc-endpoint [=arg(=127.0.0.1:8090)]
                                        Endpoint for websocket RPC to listen on
  --rpc-tls-endpoint [=arg(=127.0.0.1:8089)]
                                        Endpoint for TLS websocket RPC to 
                                        listen on
  -p [ --server-pem ] [=arg(=server.pem)]
                                        The TLS certificate file for this 
                                        server
  -P [ --server-pem-password ] arg      Password for this certificate
  --genesis-json arg                    File to read Genesis State from
  --dbg-init-key arg                    Block signing key to use for init 
                                        witnesses, overrides genesis file
  --api-access arg                      JSON file specifying API permissions
  --io-threads [=arg(=0)]               Number of IO threads, default to 0 for 
                                        auto-configuration
  --enable-subscribe-to-all [=arg(=1)]  Whether allow API clients to subscribe 
                                        to universal object creation and 
                                        removal events
  --enable-standby-votes-tracking [=arg(=1)]
                                        Whether to enable tracking of votes of 
                                        standby witnesses and committee 
                                        members. Set it to true to provide 
                                        accurate data to API clients, set to 
                                        false for slightly better performance.


  --replay-blockchain                   Rebuild object graph by replaying all 
                                        blocks without validation
  --revalidate-blockchain               Rebuild object graph by replaying all 
                                        blocks with full validation
  --resync-blockchain                   Delete all blocks and re-sync with 
                                        network from scratch
  --force-validate                      Force validation of all transactions 
                                        during normal operation
  --genesis-timestamp arg               Replace timestamp from genesis.json 
                                        with current time plus this many 
                                        seconds (experts only!)

  --p2p-endpoint arg                    Endpoint for P2P node to listen on
  -s [ --seed-node ] arg                P2P nodes to connect to on startup (may
                                        specify multiple times)
  --seed-nodes arg                      JSON array of P2P nodes to connect to 
                                        on startup
...
```

File: programs/witness_node/main.cpp [hunk 2]
```diff
@@ -73,6 +73,26 @@ int main(int argc, char** argv) {
 
       bpo::variables_map options;
 
+      bpo::options_description cli, cfg;
+      node->set_program_options(cli, cfg);
+      app_options.add(cli);
+      cfg_options.add(cfg);
+
+      if( options.count("version") )
```
Comment (abitmore) [inline programs/witness_node/main.cpp]: `options` variable is uninitialized here.

**Comment by oxarbitrage** [comment]
my apologies, i should had checked this more before submitting.

File: programs/witness_node/main.cpp (hunk 2 above)
Comment (oxarbitrage) [reply programs/witness_node/main.cpp]: done, thanks.

**Comment by oxarbitrage** [comment]
The first `app_options.add(cli); ` was removed to avoid the duplicated output of the help.

**Comment by abitmore** [comment]
Looks good. @pmconrad @jmjatlanta your opinions?

**Pull request merged**